realm/realm-library/src/objectServer/java/io/realm/mongodb/sync/SyncSession.java

```diff
@@ -259,13 +259,21 @@ public boolean isConnected() {
         return (sessionState == State.ACTIVE || sessionState == State.DYING) && connectionState == ConnectionState.CONNECTED;
     }
 
+    /**
+     * All progress listener events from native Sync are reported to this method.
+     */
+    @SuppressWarnings("unused")
     synchronized void notifyProgressListener(long listenerId, long transferredBytes, long transferableBytes) {
         Pair<ProgressListener, Progress> listener = listenerIdToProgressListenerMap.get(listenerId);
         if (listener != null) {
             Progress newProgressNotification = new Progress(transferredBytes, transferableBytes);
             if (!newProgressNotification.equals(listener.second)) {
                 listener.second = newProgressNotification;
-                listener.first.onChange(newProgressNotification);
+                try {
+                    listener.first.onChange(newProgressNotification);
+                } catch (Exception exception) {
+                    RealmLog.error(exception);
```

Comment:
This seems like a behavioral change, where we now swallow the exception?

Note, it is probably not a bad idea since callbacks generally shouldn't be allowed to crash here, but as a minimum, we would need a test for it + an update to the breaking section of the changelog

Reply:
We also swallowed it before, just in `Sync.notifyProgressListeners`. Since the callback is directly on the `SyncSession` I just replicated it. 

Maybe this was not that obvious as the `Sync.notifyProgressListener` is still there even though unused, but I left it as I thought we would/reuse it when fixing the leakage of session objects.

Reply:
The reason why I introduced it was that we in fact have a test for it :)

Reply:
Cool 👍